simpleclient/src/main/java/io/prometheus/client/Collector.java

```diff
@@ -91,24 +96,22 @@ public boolean equals(Object obj) {
           return false;
         }
         Sample other = (Sample) obj;
-        return other.name.equals(name) && other.labelNames.equals(labelNames)
-          && other.labelValues.equals(labelValues) && other.value == value;
+        return other.name.equals(name) && other.labels.equals(labels) && other.value == value;
       }
 
       @Override
       public int hashCode() {
         int hash = 1;
         hash = 37 * hash + name.hashCode();
-        hash = 37 * hash + labelNames.hashCode();
-        hash = 37 * hash + labelValues.hashCode();
+        hash = 37 * hash + labels.hashCode();
         long d = Double.doubleToLongBits(value);
         hash = 37 * hash + (int)(d ^ (d >>> 32));
         return hash;
       }
 
       @Override
       public String toString() {
-        return "Name: " + name + " LabelNames: " + labelNames + " labelValues: " + labelValues +
+        return "Name: " + name + " LabelNames: " + labels.keySet() + " labelValues: " + labels.values() +
```

Comment:
I see your point, there is no guaranteed iteration order for keySet() and values() (although if you look at the implementation in AbstractMap the entrySet() iterator is used in both cases).

A better solution could be to the change the string format and print the map as a single object.